tests/integration/test_dynamodb.py

```diff
@@ -0,0 +1,182 @@
+            'MyNumberSet': set([Decimal('1.25')]),
+            'MyBinarySet': set([Binary(b'\x01')]),
+            'MyList': ['foo'],
+            'MyMap': {'foo': 'bar'}
+        }
+        cls.table = cls.dynamodb.create_table(
+            TableName=cls.table_name,
+            ProvisionedThroughput={"ReadCapacityUnits": 5,
+                                   "WriteCapacityUnits": 5},
+            KeySchema=[{"AttributeName": "MyHashKey", "KeyType": "HASH"}],
+            AttributeDefinitions=[{"AttributeName": "MyHashKey",
+                                   "AttributeType": "S"}])
+        waiter = cls.dynamodb.meta.client.get_waiter('table_exists')
+        waiter.wait(TableName=cls.table_name)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.table.delete()
+
+
+class TestDynamoDBTypes(BaseDynamoDBTest):
+    def test_put_get_item(self):
+        self.table.put_item(Item=copy.deepcopy(self.item_data))
+        self.addCleanup(self.table.delete_item, Key={'MyHashKey': 'mykey'})
+        response = self.table.get_item(Key={'MyHashKey': 'mykey'})
+        self.assertEqual(response['Item'], self.item_data)
+
+
+class TestDynamoDBConditions(BaseDynamoDBTest):
+    @classmethod
+    def setUpClass(cls):
+        super(TestDynamoDBConditions, cls).setUpClass()
+        cls.table.put_item(Item=copy.deepcopy(cls.item_data))
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.table.delete_item(Key={'MyHashKey': 'mykey'})
+        super(TestDynamoDBConditions, cls).setUpClass()
+
+    def test_filter_expression(self):
+        response = self.table.scan(
+            FilterExpression=Attr('MyHashKey').eq('mykey'))
+        self.assertEqual(response['Count'], 1)
+
+    def test_key_condition_expression(self):
+        response = self.table.query(
+            KeyConditionExpression=Key('MyHashKey').eq('mykey'))
+        self.assertEqual(response['Count'], 1)
+
+    def test_key_condition_with_filter_condition_expression(self):
+        response = self.table.query(
+            KeyConditionExpression=Key('MyHashKey').eq('mykey'),
+            FilterExpression=Attr('MyString').eq('mystring'))
+        self.assertEqual(response['Count'], 1)
+
+    def test_condition_less_than(self):
+        response = self.table.scan(
+            FilterExpression=Attr('MyNumber').lt(Decimal('1.26')))
+        self.assertEqual(response['Count'], 1)
+
+    def test_condition_less_than_equal(self):
+        response = self.table.scan(
+            FilterExpression=Attr('MyNumber').lte(Decimal('1.26')))
+        self.assertEqual(response['Count'], 1)
+
+    def test_condition_greater_than(self):
+        response = self.table.scan(
+            FilterExpression=Attr('MyNumber').gt(Decimal('1.24')))
+        self.assertEqual(response['Count'], 1)
+
+    def test_condition_greater_than_equal(self):
+        response = self.table.scan(
+            FilterExpression=Attr('MyNumber').gte(Decimal('1.24')))
+        self.assertEqual(response['Count'], 1)
```

Comment:
Can we test the specific condition we're filtering/querying for (I'm asking for all of the tests in this class)?
For example,  `r = FilterExpression=Attr('MyNumber').gte(Decimal('1.24')))`, r['Items'].values()[0] < 1.24`.

It would put additional context in the test rather than requiring the user to scroll up and see why the count should be 1.